chore: enhance readme (#1057)
amp-labs/connectors: README.md

<div align="center">

[![Star us on GitHub](https://img.shields.io/github/stars/amp-labs/connectors?color=FFD700&label=Stars&logo=Github)](https://github.com/amp-labs/connectors) [![Discord](https://img.shields.io/badge/Join%20The%20Community-black?logo=discord)](https://discord.gg/BWP4BpKHvf) [![Documentation](https://img.shields.io/badge/Read%20our%20Documentation-black?logo=book)](https://docs.withampersand.com) ![PRs welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg) <img src="https://img.shields.io/static/v1?label=license&message=MIT&color=white" alt="License">
</div>

</div>

<br/>

<div align="center">
    <img src="https://res.cloudinary.com/dycvts6vp/image/upload/v1724756323/media/hqukkkmpk96zavslpmw5.png" alt="Ampersand Overview" width="100%">
</div>

# Overview

> [Ampersand](https://withampersand.com/trk=readme-github) is a declarative platform for SaaS builders who are creating product integrations. It allow you to:

Reviewer: ```suggestion
[Ampersand](https://withampersand.com/trk=readme-github) is a declarative platform for SaaS builders who are creating product integrations. It allows you to:
```

* Read data from your customer’s SaaS

> * Write data to your customer’s SaaS

Reviewer: We could have a line about proxy / managed auth here

* Coming soon: subscribe to events (creates, deletes, and field changes) in your customer’s SaaS

The key components include:

> - Manifest file: an amp.yaml file, where you define all your integrations: API to connect to, objects and fields you want to read or write, and configuration options you’d like to expose to your customers.

Reviewer: 1) A link to the samples repo would be nice here! 

```suggestion
- Manifest file (`amp.yaml`): Define all your integrations, the APIs to connect to, the objects and fields for reading or writing, and the configuration options you want to expose to your customers.
```

- Ampersand server: a managed service that keeps track of each of your customer’s configurations, and makes the appropriate API calls to your customer’s SaaS, while optimizing for cost, handling retries and error message parsing.

> - Embeddable UI components with Ampersand, you can embed set-up, configuration, and management UIs that allow your end users to customize and manage their integrations.

Reviewer: ```suggestion
- Embeddable UI components: You can embed set-up, configuration, and management UIs that allow your end users to customize and manage their integrations.
```

> - Dashboard: our dashboard allows you to monitor and troubleshoot your customers’ integrations.

Reviewer: ```suggestion
- Dashboard: Provides deep observability into customer integrations, allowing you to monitor & troubleshoot with logs
```
